packages/sdk/src/client.ts

```diff
@@ -647,6 +652,48 @@ export function SoundClient({
     return mintSchedules.flat().sort((a, b) => a.startTime - b.startTime)
   }
 
+  async function getContractError(txHash: string) {
+    if (
+      txHash === NULL_BYTES32 ||
+      txHash.slice(0, 2) !== '0x' ||
+      // Tx hash is 32 bytes, which is 64 hex characters + '0x'
+      txHash.length !== 66
+    ) {
+      throw new InvalidTxHashError({ txHash })
+    }
+
+    const provider = await _requireProvider()
+    const tx = await provider.getTransaction(txHash)
+
+    try {
+      // Simulate the original transaction
+      const response = await provider.call(
+        {
+          to: tx.to,
+          from: tx.from,
+          nonce: tx.nonce,
+          gasLimit: tx.gasLimit,
+          gasPrice: tx.gasPrice,
+          data: tx.data,
+          value: tx.value,
+          chainId: tx.chainId,
+          type: tx.type ?? undefined,
+          accessList: tx.accessList,
+        },
+        tx.blockNumber,
+      )
+
+      // If this is a failed transaction, the first 4 bytes of the response
+      // will be the custom error selector (hash of its signature)
+      const firstFourBytes = response.slice(0, 10)
+
+      return errorSigHashToName[firstFourBytes] ?? 'Unable to parse error'
+    } catch (err) {
+      console.error(err)
+    }
+    return
```

Comment:
let's improve the type-signature, the idea is to make the usage of this helper clear

Reply:
this includes: explicit type for what's to be returned, just `string | undefined` is not good, and also `null > undefined`

Reply:
This look ok? https://github.com/soundxyz/sdk/pull/175/commits/bd9224514d344871a92c07c742c40e4e5d17c297